src/System.Linq.Parallel/src/System/Linq/Parallel/Enumerables/EmptyEnumerable.cs

```diff
@@ -64,8 +65,9 @@ internal override bool MoveNext(ref T currentElement, ref int currentKey)
         }
 
         // IEnumerator<T> methods.
-        public T Current { get { return default(T); } }
-        object IEnumerator.Current { get { return null; } }
+        [MaybeNull]
```

Comment:
❕ This method implements `IEnumerator<T>.Current`. It is not valid to widen the return type by adding `[MaybeNull]` here.

Reply:
By my understanding even when T is non nullable reference type default(T) will be null, so added this annotation. But your point makes sense, I guess I should remove it